message/validation/errors.go

```diff
@@ -0,0 +1,100 @@
+package validation
+
+import (
+	"fmt"
+	"strings"
+)
+
+type Error struct {
```

Comment:
error usually contains the text inside, not sure why you need both innerErr and text. you also just use composition and put error inside.
I understand how you use `Error()` here in tests to print stuff pretty, but this could be achieved by a test helper 

func ErrorMessage(e Error) string {
	var sb strings.Builder
	sb.WriteString(e.Error())

	if e.got != nil {
		sb.WriteString(fmt.Sprintf(", got %v", e.got))
	}
	if e.want != nil {
		sb.WriteString(fmt.Sprintf(", want %v", e.want))
	}
        // this is not needed since error will always be assigned as its the new `text`
	//if e.innerErr != nil {
	//       sb.WriteString(fmt.Sprintf(": %s", e.innerErr.Error()))
	//}

	return sb.String()
}

Reply:
Also given we put hex encoded signature/PK into the error (I replaced them with `...` in the example above), we'd have a separate error shown in Grafana for each signature/PK, so we can easily have hundreds of them, which would make the dashboard useless